src/modm/driver/dac/cs43l22_impl.hpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2021, Marton Ledneczki
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_CS43L22_HPP
+#	error "Don't include this file directly, use 'cs43l22.hpp' instead!"
+#endif
+
+namespace modm
+{
+
+template<typename I2cMaster, typename I2sMaster>
+CS43L22<I2cMaster, I2sMaster>::CS43L22(uint8_t i2cAddress)
+	: I2cDevice<I2cMaster, 3>(i2cAddress) {}
+
+template<typename I2cMaster, typename I2sMaster>
+ResumableResult<bool>
+CS43L22<I2cMaster, I2sMaster>::initialize()
+{
+	RF_BEGIN();
+	// Verify the chip ID
+	success = RF_CALL_BLOCKING(readRegister(Register::ChipIdRevision));
```

Comment:
Just `RF_CALL` otherwise it's a synchronous blocking call.

Reply:
I replaced all the blocking calls. It took me a bit to get my head around resumable functions.